File: 2015/go/day02/part2.go
package main

import (
	"sort"
	"strconv"
	"strings"
)

func processPart2(input string) int {
	lines := strings.Split(input, "\n")

	total := 0
	for _, line := range lines {
		sides := strings.Split(line, "x")

		l, _ := strconv.Atoi(sides[0])
		w, _ := strconv.Atoi(sides[1])
		h, _ := strconv.Atoi(sides[2])

		sortedSides := []int{l, w, h}
		sort.Ints(sortedSides)

		ribbon := 2*sortedSides[0] + 2*sortedSides[1]
		bow := l * w * h

		total += ribbon + bow
	}
	return total
}

File: 2015/deno/day02/part-2.ts
export function processPart2(input: string): number {
	const trimmed = input.trim();
	const lines = trimmed.split('\n');

	let ribbon = 0;

	for (const line of lines) {
		const [l, w, h] = line.split('x').map(Number);
		const sorted = [l, w, h].sort((a, b) => a - b);
		const [shortest, middle] = sorted;

		ribbon += 2 * shortest + 2 * middle + l * w * h;
	}
	return ribbon;
}
